deploy/lib/xquery/setup.xqy

```diff
@@ -3136,6 +3140,59 @@ declare function setup:validate-group(
       setup:validation-fail(fn:concat("Missing Group: ", $group))
 };
 
+declare function setup:validate-group-settings(
+  $group-config as element(gr:group),
+  $setting as element()
+) {
+  let $group := $group-config/gr:group-name/fn:string(.)
+  let $setting-name := $setting/fn:string(.)
+  let $function-name := fn:concat("group-get-", $setting-name, "s")
+  let $keys := fn:tokenize($setting/@keys, " ")
+  let $existing :=
+    try
+    {
+      xdmp:eval(
```

Comment:
Thanks for pointing that out. Took a quick look and saw the same. As you probably know, eval is slow and we should avoid using it unnecessarily wherever possible. CC @dmcassel @grtjn in case there's a reason to keep it that I'm not aware of.

Beyond this, your code was a pleasure to read through and I was quite impressed by the gymnastics you performed in lieu of missing admin:group-set* functions as well as complicated string concatenations you performed to make this code forward compatible with new versions of ML with little development effort. I'd have no problem merging this in with the eval removed.